File: 506-relative-ranks/relative-ranks.java
import java.util.Arrays;

class Solution {
    public String[] findRelativeRanks(int[] score) {
        Integer[] indexes = new Integer[score.length];
        for (int i = 0; i < score.length; i++) {
            indexes[i] = i;
        }

        Arrays.sort(indexes, (a, b) -> Integer.compare(score[b], score[a]));

        String[] result = new String[score.length];
        for (int i = 0; i < score.length; i++) {
            if (i == 0) {
                result[indexes[i]] = "Gold Medal";
            } else if (i == 1) {
                result[indexes[i]] = "Silver Medal";
            } else if (i == 2) {
                result[indexes[i]] = "Bronze Medal";
            } else {
                result[indexes[i]] = String.valueOf(i + 1);
            }
        }

        return result;
    }
}

File: 506-relative-ranks/relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:


        sorted_score =[(num,i) for i,num in enumerate(score)]

        sorted_score.sort(reverse = True)
 
        for i in range(len(score)):
            num ,ind = sorted_score[i]
            if i==0:
                score[ind] = "Gold Medal"
            elif i==1:
                score[ind] = "Silver Medal"
            elif i ==2:
                score[ind] = "Bronze Medal"
            else:
                score[ind] =str(i+1)
        return score

            
        